digital_land/repository/entry_repository.py

```diff
@@ -0,0 +1,198 @@
+        "returns all Entries associated with the specificed entity ref"
+
+        cursor = self.conn.cursor()
+        cursor.execute(
+            """
+            SELECT
+                fact.*,
+                entry.resource AS "__RESOURCE__",
+                entry.line_num AS "__LINE_NUM__"
+            FROM fact
+            JOIN provenance ON provenance.fact = fact.id
+            JOIN entry ON provenance.entry = entry.id
+            WHERE entry.entity = ?
+        """,
+            (entity,),
+        )
+        return self._entries_from_rows(entity, cursor.fetchall())
+
+    def find_by_fact(self, fact):
+        "returns all Entries that state the given fact"
+
+        cursor = self.conn.cursor()
+        cursor.execute(
+            """
+            SELECT
+                fact.*,
+                entry.resource AS "__RESOURCE__",
+                entry.line_num AS "__LINE_NUM__"
+            FROM fact
+            JOIN provenance ON provenance.fact = fact.id
+            JOIN entry ON provenance.entry = entry.id
+            WHERE fact.entity = ?
+            AND fact.attribute = ?
+            AND fact.value = ?
+        """,
+            (fact.entity, fact.attribute, fact.value),
+        )
+
+        result = self._entries_from_rows(fact.entity, cursor.fetchall())
+        return result
+
+    def _create_schema(self):
+        cursor = self.conn.cursor()
+        cursor.execute("""PRAGMA foreign_keys = ON""")
+
+        logger.warning("dropping tables")
+        cursor.execute("""DROP TABLE IF EXISTS provenance""")
+        cursor.execute("""DROP TABLE IF EXISTS fact""")
+        cursor.execute("""DROP TABLE IF EXISTS entry""")
+        cursor.execute("""DROP TABLE IF EXISTS entity""")
+
+        logger.warning("creating tables")
+        cursor.execute(
+            """CREATE TABLE entity (slug TEXT PRIMARY KEY NOT NULL CHECK(LENGTH(slug) > 0))"""
+        )
+        cursor.execute(
+            """CREATE TABLE entry (
+                id INTEGER PRIMARY KEY,
+                resource TEXT,
+                line_num INTEGER,
+                entity NOT NULL,
+                FOREIGN KEY(entity) REFERENCES entity(slug),
+                UNIQUE(resource, line_num)
+            )"""
+        )
+        cursor.execute(
+            """CREATE TABLE fact (
+                id INTEGER PRIMARY KEY,
+                entity TEXT,
+                attribute TEXT,
+                value TEXT,
+                entry_date,
+                start_date,
+                end_date,
+                FOREIGN KEY(entity) REFERENCES entity(slug),
+                UNIQUE(entity, attribute, value)
+            )"""
+        )
+        cursor.execute(
+            """CREATE TABLE provenance (
+                entry INTEGER,
+                fact INTEGER,
+                entry_date,
+                start_date,
+                end_date,
+                FOREIGN KEY(entry) REFERENCES entry(id),
+                FOREIGN KEY(fact) REFERENCES fact(id)
```

Comment:
Yeah maybe; it won't currently make any difference as we always throw the whole thing away and rebuild it, and the nature of it means that it will never clash on provenance. If and when we start updating things in place, this will be more important.